src/Collection/Collection.service.ts

```diff
@@ -23,18 +32,210 @@ import {
+    dbCollection: ThirdPartyCollectionAttributes,
+    dbItems: ItemAttributes[],
+    signedMessage: string,
+    signature: string
+  ): Promise<{
+    collection: CollectionAttributes
+    items: FullItem[]
+  }> {
+    // For DCL collections, once a published collection item changes, the PUSH CHANGES button appears
+    // That will fire a /collections/${collectionId}/curation which will create a new CollectionCuration
+    // Subsequent changes will not show the push changes button, as it's already under_review
+
+    // For TP items, creations always exist. PUSH CHANGES should appear if the item has an approved ItemCuration and has changes in the Catalyst
+    // That should fire /items/:id/curation for each item that changed
+
+    // What we should migrate over here is the creation of the virtual CollectionCuration, as there'll always be a publish before a PUSH CHANGES
+
+    if (dbItems.length === 0) {
+      throw new InvalidRequestError('Tried to publish no TP items')
+    }
+
+    try {
+      ethers.utils.verifyMessage(signedMessage, signature) // Throws if invalid
+    } catch (error) {
+      throw new InvalidRequestError(
+        'Tried to publish TP items with an invalid signed message or signature'
+      )
+    }
+
+    const collectionId = dbItems[0].collection_id
+    if (!collectionId) {
+      throw new InvalidRequestError(
+        'Cannot publish items without a collection id'
+      )
+    }
+
+    const allTheSameCollection = dbItems.every(
+      (item) => item.collection_id === collectionId
+    )
+    if (!allTheSameCollection) {
+      throw new InvalidRequestError(
+        'Cannot publish items that belong to different collections'
+      )
+    }
+
+    const isPublished = await ItemCuration.findLastCreatedByCollectionIdAndStatus(
+      collectionId,
+      CurationStatus.PENDING
+    )
+    if (isPublished) {
+      throw new AlreadyPublishedCollectionError(
+        collectionId,
+        CollectionType.THIRD_PARTY,
+        CollectionAction.UPDATE
+      )
+    }
+
+    const now = new Date()
+    await SlotUsageCheque.create<SlotUsageChequeAttributes>({
+      id: uuid(),
+      signedMessage,
+      collection_id: dbCollection.id,
+      third_party_id: dbCollection.third_party_id,
+      created_at: now,
+      updated_at: now,
+    })
+
+    const promises = []
+    for (const item of dbItems) {
+      promises.push(
+        ItemCuration.create<ItemCurationAttributes>({
+          id: uuid(),
+          item_id: item.id,
+          status: CurationStatus.PENDING,
+          created_at: now,
+          updated_at: now,
+        })
+      )
+    }
+    const itemCurations = await Promise.all(promises)
+    const [lastItemCuration] = itemCurations.slice(-1)
+
+    const collectionCuration = await CollectionCuration.findOne(collectionId)
+    if (collectionCuration) {
+      await CollectionCuration.update(
+        { id: collectionCuration.id },
+        { updated_at: now }
+      )
```

Comment:
If I'm not wrong, we're not updating the `CollectionCuration` whenever we update an ItemCuration in the `Curations.router`, should we move this logic there as well?

Reply:
Very possible, I'll check and add it or at least an issue